### ISSUE TITLE
Problem when the loss value changes to nan at some point.

### ISSUE THREAD

**Issue by haaaaak** [ultralytics/ultralytics#4733]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello. I was doing research with segmentation model as usual.
But from what point did all the losses proceed with nan values, and for what reason do this problem occur?
Normally, this kind of problem never happened.
I only changed the optimizer from auto(SGD) to adamW.
And this happened and I couldn't find an answer to the related problem.

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    110/500      31.2G     0.6495     0.8474     0.3936     0.8768        156       1024: 100%|██████████| 1377/1377 [05:09<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.962      0.963      0.984      0.883      0.486      0.464      0.414      0.302

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    111/500      30.1G     0.6464     0.8513     0.3938     0.8755         22       1024: 100%|██████████| 1377/1377 [04:54<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.962      0.953      0.982      0.885      0.456      0.454      0.404      0.298

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    112/500      31.8G     0.6514     0.8567     0.3937     0.8763        236       1024: 100%|██████████| 1377/1377 [06:49<00:00,  3.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.949      0.962      0.981      0.886      0.446      0.447      0.406      0.298

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    113/500      34.4G     0.6431     0.8413     0.3897     0.8738         65       1024: 100%|██████████| 1377/1377 [05:34<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.951      0.964      0.981      0.883      0.497      0.482      0.444      0.318

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    114/500      38.6G     0.6401     0.8412     0.3881     0.8737        109       1024: 100%|██████████| 1377/1377 [05:06<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.959      0.959      0.982      0.885       0.48      0.483      0.436      0.311

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    115/500      29.2G     0.6421      0.833     0.3892      0.876         91       1024: 100%|██████████| 1377/1377 [04:52<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.951      0.961      0.982      0.889      0.472      0.472      0.427      0.307

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    116/500      19.4G     0.6437        nan     0.3893     0.8732         24       1024: 100%|██████████| 1377/1377 [04:34<00:00,  5.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.957      0.961      0.982       0.89          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    117/500      25.3G     0.6399        nan     0.3855     0.8729        107       1024: 100%|██████████| 1377/1377 [04:50<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.962      0.956      0.981      0.889          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    118/500      31.7G     0.6421     0.8436     0.3867     0.8745        242       1024: 100%|██████████| 1377/1377 [05:26<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.958      0.962      0.983      0.883          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    119/500        27G     0.6436     0.8471      0.389     0.8733         59       1024: 100%|██████████| 1377/1377 [04:41<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.949      0.958      0.977      0.877          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    120/500        34G     0.6445        nan     0.3872     0.8728        123       1024: 100%|██████████| 1377/1377 [05:44<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.965      0.953      0.984      0.883          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    121/500      31.2G      0.654        nan     0.3934      0.875        121       1024: 100%|██████████| 1377/1377 [04:51<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.963      0.959      0.982      0.881          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    122/500      26.8G     0.6585        nan     0.3931     0.8741        113       1024: 100%|██████████| 1377/1377 [04:44<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.952      0.957      0.981      0.881          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    123/500      25.7G     0.6613        nan      0.395     0.8749         71       1024: 100%|██████████| 1377/1377 [04:44<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.951      0.955       0.98      0.877          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    124/500      19.5G     0.6607        nan     0.3951     0.8752        171       1024: 100%|██████████| 1377/1377 [04:30<00:00,  5.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.963      0.953       0.98      0.874          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
    125/500      29.9G     0.6574        nan     0.3945     0.8751         20       1024: 100%|██████████| 1377/1377 [05:28<00:00,  4.
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95):
                   all       1375      50347      0.965      0.954       0.98      0.873          0          0          0          0

...

In the early days of learning, nan did not occur.
However, when a certain amount of learning was reached, all loss values were returned to nan, starting with seg_loss, and both train and validation were returned to nan.
Why is it doing this?

In addition, the train command was entered as follows.
```
from ultralytics import YOLO
model = YOLO("yolov8-seg.yaml")
model = YOLO("yolov8m-seg.pt")

model.train(data="data.yaml", imgsz=1024, batch=4, epochs=500, patience=500, optimizer='AdamW')
```

**Comment by TyrionChou** [comment]
hi~，have you solve the problem？

**Comment by glenn-jocher** [comment]
@TyrionChou hello! We're currently working on resolving the issue you mentioned. Your patience is greatly appreciated. We understand how important this is and we're committed to providing a solution. We'll update you as soon as we have any new developments. Thank you for bringing this to our attention.

**Comment by mkewsds12** [comment]
```
  Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss   sem_loss  Instances       Size
 28/250      6.07G    0.04431        nan     0.5388      1.283          0          2        640: 100% ━━━━━━━━━━━━ 1802/1802 3.7it/s 8:13
             Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 258/258 3.8it/s 1:08
               all       8234       8503      0.649      0.686      0.682      0.667          0          0          0          0

  Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss   sem_loss  Instances       Size
 29/250      6.07G        nan        nan        nan        nan          0          3        640: 100% ━━━━━━━━━━━━ 1802/1802 3.7it/s 8:12
             Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 258/258 4.5it/s 57.8s
               all       8234       8503          0          0          0          0          0          0          0          0
```
```
Transferred 555/561 items from pretrained weights
New [https://pypi.org/project/ultralytics/8.4.21](vscode-file://vscode-app/e:/Users/EDY/AppData/Local/Programs/Microsoft%20VS%20Code/ce099c1ed2/resources/app/out/vs/code/electron-browser/workbench/workbench.html) available Update with 'pip install -U ultralytics'
WARNING 'label_smoothing' is deprecated and will be removed in the future.
Ultralytics 8.4.11 Python-3.12.12 torch-2.9.1+cu128 CUDA:0 (NVIDIA GeForce RTX 5070, 12227MiB)
engine\trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=2.0, cache=disk, cfg=None, classes=None, close_mosaic=50, cls=1.0, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=True, cutmix=0.0, data=model\20250711.yaml, degrees=10.0, deterministic=True, device=0, dfl=2.0, dnn=False, dropout=0.05, dynamic=False, embed=None, end2end=None, epochs=250, erasing=0.0, exist_ok=True, fliplr=0.5, flipud=0.3, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.3, hsv_s=0.3, hsv_v=0.6, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.001, lrf=0.05, mask_ratio=2, max_det=300, mixup=0.0, mode=train, model=model\yolo11s-seg.yaml, momentum=0.937, mosaic=0.0, multi_scale=False, name=yolo11_training_20260313_102100, nbs=64, nms=False, opset=None, optimize=False, optimizer=AdamW, overlap_mask=True, patience=15, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=training_output, rect=False, resume=False, retina_masks=False, rle=1.0, save=True, save_conf=False, save_crop=False, save_dir=E:\landfun\code\muke\20260309-\runs\segment\training_output\yolo11_training_20260313_102100, save_frames=False, save_json=False, save_period=10, save_txt=False, scale=0.5, seed=42, shear=2.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=segment, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=5, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None

0 -1 1 928 ultralytics.nn.modules.conv.Conv [3, 32, 3, 2]
1 -1 1 18560 ultralytics.nn.modules.conv.Conv [32, 64, 3, 2]
2 -1 1 26080 ultralytics.nn.modules.block.C3k2 [64, 128, 1, False, 0.25]
3 -1 1 147712 ultralytics.nn.modules.conv.Conv [128, 128, 3, 2]
4 -1 1 103360 ultralytics.nn.modules.block.C3k2 [128, 256, 1, False, 0.25]
5 -1 1 590336 ultralytics.nn.modules.conv.Conv [256, 256, 3, 2]
6 -1 1 346112 ultralytics.nn.modules.block.C3k2 [256, 256, 1, True]
7 -1 1 1180672 ultralytics.nn.modules.conv.Conv [256, 512, 3, 2]
8 -1 1 1380352 ultralytics.nn.modules.block.C3k2 [512, 512, 1, True]
9 -1 1 656896 ultralytics.nn.modules.block.SPPF [512, 512, 5]
10 -1 1 990976 ultralytics.nn.modules.block.C2PSA [512, 512, 1]
11 -1 1 0 torch.nn.modules.upsampling.Upsample [None, 2, 'nearest']
12 [-1, 6] 1 0 ultralytics.nn.modules.conv.Concat [1]
13 -1 1 443776 ultralytics.nn.modules.block.C3k2 [768, 256, 1, False]
14 -1 1 0 torch.nn.modules.upsampling.Upsample [None, 2, 'nearest']
15 [-1, 4] 1 0 ultralytics.nn.modules.conv.Concat [1]
16 -1 1 127680 ultralytics.nn.modules.block.C3k2 [512, 128, 1, False]
17 -1 1 147712 ultralytics.nn.modules.conv.Conv [128, 128, 3, 2]
18 [-1, 13] 1 0 ultralytics.nn.modules.conv.Concat [1]
19 -1 1 345472 ultralytics.nn.modules.block.C3k2 [384, 256, 1, False]
20 -1 1 590336 ultralytics.nn.modules.conv.Conv [256, 256, 3, 2]
21 [-1, 10] 1 0 ultralytics.nn.modules.conv.Concat [1]
22 -1 1 1511424 ultralytics.nn.modules.block.C3k2 [768, 512, 1, True]
23 [16, 19, 22] 1 1475065 ultralytics.nn.modules.head.Segment [3, 32, 128, 16, None, [128, 256, 512]]
YOLO11s-seg summary: 204 layers, 10,083,449 parameters, 10,083,433 gradients

Transferred 561/561 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed
train: Fast image access (ping: 0.10.1 ms, read: 170.8147.3 MB/s, size: 57.6 KB)
train: Scanning data3\train\labels.cache... 24609 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 24609/24609 0.0s
train: Caching images (29.1GB Disk): 100% ━━━━━━━━━━━━ 24609/24609 65.2Kit/s 0.4s
val: Fast image access (ping: 0.20.1 ms, read: 101.659.6 MB/s, size: 48.5 KB)
val: Scanning data3\valid\labels.cache... 8203 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 8203/8203 0.0s
val: Caching images (9.7GB Disk): 100% ━━━━━━━━━━━━ 8203/8203 59.3Kit/s 0.1s
Plotting labels to E:\landfun\code\muke\20260309-\runs\segment\training_output\yolo11_training_20260313_102100\labels.jpg...
optimizer: AdamW(lr=0.001, momentum=0.937) with parameter groups 90 weight(decay=0.0), 101 weight(decay=0.0005), 100 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to E:\landfun\code\muke\20260309-\runs\segment\training_output\yolo11_training_20260313_102100
Starting training for 250 epochs...
```
The same problem occurred during my training today, even though the same dataset had no issues before, and I hadn't even modified any parameters. I'm using yolo11s-seg.

**Comment by glenn-jocher** [comment]
Thanks for the report — since this is a closed stale issue and your case is a newer `yolo11s-seg` run, please open a new issue for visibility and include `yolo checks` (or `ultralytics.utils.checks.collect_system_info()` if you suspect multiple installs), the full text traceback, and ideally a small reproducible dataset sample; also please test on the latest Ultralytics version first, as newer trainers include NaN recovery logic in [`BaseTrainer._handle_nan_recovery()`](https://docs.ultralytics.com/reference/engine/trainer/) that may help diagnose or recover from this case.
